lib/formats/odata.js

```diff
@@ -510,7 +510,7 @@ const singleRowToOData = (fields, row, domain, originalUrl, query) => {
     let { offset } = paging;
 
     if (skipToken) {
-      offset = filtered.fiindIndex(s => skipToken.repeatId === s.__id);
+      offset = filtered.findIndex(s => skipToken.repeatId === s.__id);
```

Comment:
This is the source of the confusion and the bug, we need to send repeats after the one in `skipToken`.

```suggestion
      offset = filtered.findIndex(s => skipToken.repeatId === s.__id) + 1;
```